sfge/src/main/java/com/salesforce/graph/ops/expander/ApexPathExpander.java

```diff
@@ -177,13 +178,15 @@ final class ApexPathExpander implements ClassStaticScopeProvider, EngineDirectiv
 
     ApexPathExpander(
             GraphTraversalSource g,
-            ApexPathCollapser apexPathCollapser,
+            Long pathExpansionId,
             ApexPath topMostPath,
-            ApexPathExpanderConfig config) {
+            ApexPathExpanderConfig config,
+            PathExpansionRegistry registry) {
         this.id = ID_GENERATOR.incrementAndGet();
         this.hash = Objects.hashCode(this.id);
         this.g = g;
-        this.apexPathCollapser = apexPathCollapser;
+        this.pathExpansionId = pathExpansionId;
+        this.registry = registry;
         this.forkEvents = new LinkedHashMap<>();
         this.forkResults = new HashMap<>();
```

Comment:
I had some more changes where I tried to move these two fields to use the registry as well along with making `PathVertex` an indexable class. Been hitting a bug with that and decided to hold off for now.